### PR TITLE
#6311 Optional Mod Integration Configs during activation

### LINKED ISSUE #6311: Slice 4.2: support optional integration configurations during activation
* https://www.notion.so/pixiebrix/Make-Google-and-Google-Sheets-a-Standard-Integration-efb28821a2b54365a85bcd6981e9c71c?pvs=4#9ab5b6d68cff418cb5b85f5764625604

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Closes #6311 

## Reviewer Tips

- There is still no page editor support, mods need to be edited in the workshop to use this feature

## Remaining work

- [x] Figure out issue with try-except brick not catching the optional integration error
- [x] Test other activation flows outside of console recipe activation, test cloud extension and sidebar activation
- [x] Write some unit tests for mod activation with optional integrations

## Checklist

- [x] Add tests
- [ ] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer - @grahamlangford

### CODE REVIEW

**Comment by twschiller** [comment]
Link to Slack conversation here: https://pixiebrix.slack.com/archives/C05M944NW3Y/p1699361784967569?thread_ts=1699323634.976269&cid=C05M944NW3Y

Includes my theory for what's broken: https://pixiebrix.slack.com/archives/C05M944NW3Y/p1699361784967569?thread_ts=1699323634.976269&cid=C05M944NW3Y

File: src/integrations/util/makeServiceContextFromDependencies.ts [hunk 1]
```diff
@@ -53,10 +53,14 @@ export default async function makeServiceContextFromDependencies(
...
     Object.fromEntries(
-      dependencies.map((dependency) => [
-        `@${dependency.outputKey}`,
-        dependencyContext(dependency),
-      ])
+      dependencies.map((dependency) => {
+        const context =
+          dependency.isOptional && !dependency.configId
+            ? null
+            : dependencyContext(dependency);
```
Comment (grahamlangford) [inline src/integrations/util/makeServiceContextFromDependencies.ts:60]: NIT: I think it's more readable to have `null` be the falsy path.

```suggestion
        const context =
          dependency.configId && !dependency.isOptional
            ? dependencyContext(dependency)
            : null;
```

File: src/sidebar/activateRecipe/ActivateModPanel.test.tsx
Comment (grahamlangford) [inline src/sidebar/activateRecipe/ActivateModPanel.test.tsx:1]: NIT: I thought we were moving away from relying mainly on snapshot testing?

Comment (grahamlangford) [reply src/sidebar/activateRecipe/ActivateModPanel.test.tsx:1]: We could check for the correct text and buttons instead.

**Review by grahamlangford — APPROVED** [review]

File: src/sidebar/activateRecipe/ActivateModPanel.test.tsx
Comment (BLoe) [reply src/sidebar/activateRecipe/ActivateModPanel.test.tsx:1]: CoPilot generated these tests for me instantly based on the rest of the file using snapshots. But yeah I can change to text assertions

**Pull request merged**